src/layout/layout.rs

```diff
@@ -41,6 +41,68 @@ thread_local! {
     ));
 }
 
+/// Represents the spacing between segments in a layout.
+///
+/// The `Spacing` enum is used to define the spacing between segments in a layout. It can represent
+/// either positive spacing (space between segments) or negative spacing (overlap between segments).
+///
+/// # Variants
+///
+/// - `Space(u16)`: Represents positive spacing between segments. The value indicates the number of
+///   cells.
+/// - `Overlap(u16)`: Represents negative spacing, causing overlap between segments. The value
+///   indicates the number of overlapping cells.
+///
+/// # Default
+///
+/// The default value for `Spacing` is `Space(0)`, which means no spacing or no overlap between
+/// segments.
+///
+/// # Conversions
+///
+/// The `Spacing` enum can be created from different integer types:
+///
+/// - From `u16`: Directly converts the value to `Spacing::Space`.
+/// - From `i16`: Converts negative values to `Spacing::Overlap` and non-negative values to
+///   `Spacing::Space`.
+/// - From `i32`: Clamps the value to the range of `i16` and converts negative values to
+///   `Spacing::Overlap` and non-negative values to `Spacing::Space`.
+///
+/// See the [`Layout::spacing`] method for details on how to use this enum.
+#[derive(Debug, Clone, Eq, PartialEq, Hash)]
+pub enum Spacing {
```

Comment:
needs to be re-exported in src/layout.rs

Reply:
Resolved in a0308d68eaf6